spec/truffle/interop/special_forms_spec.rb

```diff
@@ -65,11 +65,57 @@
 
   it description['[index]', :readArrayElement, [:index]] do
     pfo, pa, l  = proxy[TruffleInteropSpecs::PolyglotArray.new]
-    -> { pfo[0] }.should raise_error(IndexError)
+    pfo[0] = 1
+    pfo[0]
+    l.log.should include(['readArrayElement', 0])
+    l.log.should include(['getArraySize'])
+    l.log.should include(['hasArrayElements'])
+    pa.log.should include([:polyglot_array_size])
+    pa.log.should include([:polyglot_has_array_elements?])
+    pa.log.should include([:polyglot_array_element_readable?, 0])
+    pa.log.should include([:polyglot_read_array_element, 0])
```

Comment:
hasArrayElements/:polyglot_has_array_elements? is only used if assertions are enabled, we should not check for it here.
Same for `polyglot_array_element_readable?`.
The main messages here are readArrayElement & getArraySize, we should not check for other messages.

Same for other new methods.